backend/service/service_account.go

```diff
@@ -84,3 +100,87 @@ func isProfiler(address string) bool {
 	}
 	return false
 }
+
+func (service *AccountService) QueryDelegations(address string) (result []*vo.AccountDelegationsVo) {
+	delegations := lcd.GetDelegationsByDelAddr(address)
+	result = make([]*vo.AccountDelegationsVo, 0, len(delegations))
+	for _, val := range delegations {
+		data := vo.AccountDelegationsVo{
+			Address: val.ValidatorAddr,
+			Shares:  val.Shares,
+			Height:  val.Height,
+		}
+		valdator, err := document.Validator{}.QueryValidatorDetailByOperatorAddr(val.ValidatorAddr)
+		if err == nil {
+			data.Moniker = valdator.Description.Moniker
+			data.Amount = computeVotingPower(valdator, val.Shares)
+		}
+		result = append(result, &data)
+	}
+
+	return result
+}
+
+func computeVotingPower(validator document.Validator, shares string) utils.Coin {
+	rate, err := utils.QuoByStr(validator.Tokens, validator.DelegatorShares)
+	if err != nil {
+		logger.Error("validator.Tokens / validator.DelegatorShares", logger.String("err", err.Error()))
+		rate, _ = new(big.Rat).SetString("1")
+	}
+	sharesAsRat, ok := new(big.Rat).SetString(shares)
+	if !ok {
+		logger.Error("convert validator.Tokens type (string to big.Rat) ", logger.Any("result", ok),
+			logger.String("validator tokens", validator.Tokens))
+	}
+
+	tokensAsRat := new(big.Rat)
+	tokensAsRat.Mul(rate, sharesAsRat)
+	amount, _ := strconv.ParseFloat(tokensAsRat.FloatString(4), 64)
+
+	return utils.Coin{
+		Amount: amount,
+		Denom:  constant.IrisAttoUnit,
+	}
+}
+
+func (service *AccountService) QueryUnbondingDelegations(address string) (result []*vo.AccountUnbondingDelegationsVo) {
+
+	unbondingdelegations := lcd.GetUnbondingDelegationsByDelegatorAddr(address)
+
+	for _, val := range unbondingdelegations {
+		data := vo.AccountUnbondingDelegationsVo{
+			Address: val.ValidatorAddr,
+			EndTime: val.MinTime,
+			Height:  val.CreationHeight,
+			Amount:  utils.ParseCoin(val.Balance),
+		}
+		valdator, err := document.Validator{}.QueryValidatorDetailByOperatorAddr(val.ValidatorAddr)
```

Comment:
don't execute db query in a loop, you can query validators by operator address list